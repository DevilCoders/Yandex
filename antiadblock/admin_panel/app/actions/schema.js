import serviceApi from 'app/api/service';
import {setGlobalErrors} from 'app/actions/errors';

export const START_SCHEMA_LOADING = 'START_SCHEMA_LOADING';
export const END_SCHEMA_LOADING = 'END_SCHEMA_LOADING';

export function fetchSchema(labelId) {
    return dispatch => {
        dispatch(startSchemaLoading(labelId));

        return serviceApi.fetchSchema(labelId).then(schema => {
            return dispatch(endSchemaLoading(labelId, schema));
        }, error => {
            return dispatch(setGlobalErrors([error.message]));
        });
    };
}

export function startSchemaLoading(labelId) {
    return {
        type: START_SCHEMA_LOADING,
        labelId
    };
}

export function endSchemaLoading(labelId, schema) {
    return {
        type: END_SCHEMA_LOADING,
        labelId,
        schema
    };
}
