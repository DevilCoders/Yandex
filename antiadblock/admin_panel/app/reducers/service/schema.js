import * as serviceActions from 'app/actions/service';
import * as schemaActions from 'app/actions/schema';
import update from 'app/lib/update';

export default function(state, action) {
    const {type} = action;
    const initialState = {
        schema: [],
        loaded: false
    };

    state = state || initialState;

    switch (type) {
        case serviceActions.START_SERVICE_LOADING:
        case schemaActions.START_SCHEMA_LOADING: {
            return update(state, {
                $set: initialState
            });
        }
        case schemaActions.END_SCHEMA_LOADING: {
            return update(state, {
                $set: {
                    schema: [...action.schema],
                    loaded: true
                }
            });
        }
        default: {
            return state;
        }
    }
}

