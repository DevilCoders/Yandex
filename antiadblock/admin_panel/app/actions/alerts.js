import servicesApi from 'app/api/services';
import {setErrors, clearErrors} from 'app/actions/errors';
import {getServices, getErrors} from 'app/reducers/index';
import {getById} from 'app/reducers/errors';
import {timeIsOver} from 'app/lib/errors-logic';

export const START_ALERTS_LOADING = 'START_ALERTS_LOADING';
export const END_ALERTS_LOADING = 'END_ALERTS_LOADING';

export function fetchAlerts() {
    return (dispatch, getState) => {
        dispatch(startAlertsLoading());

        return servicesApi.fetchAlerts().then(alerts => {
            const errors = getById(getErrors(getState()), 'dashboard');
            if (errors && errors.length) {
                dispatch(clearErrors('dashboard'));
            }

            return dispatch(endAlertsLoading(alerts.items));
        }, error => {
            const lastUpdateTime = getServices(getState()).lastUpdateTimeAlerts;

            if (timeIsOver(lastUpdateTime)) {
                return dispatch(setErrors('dashboard', [error.message]));
            }
        });
    };
}

export function startAlertsLoading() {
    return {
        type: START_ALERTS_LOADING
    };
}

export function endAlertsLoading(items) {
    return {
        type: END_ALERTS_LOADING,
        items
    };
}
