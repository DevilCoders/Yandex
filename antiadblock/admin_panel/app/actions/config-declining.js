export const OPEN_CONFIG_DECLINING = 'OPEN_CONFIG_DECLINING';
export const CLOSE_CONFIG_DECLINING = 'CLOSE_CONFIG_DECLINING';

export function openConfigDeclining(serviceId, labelId, id) {
    return {
        type: OPEN_CONFIG_DECLINING,
        serviceId,
        labelId,
        id
    };
}

export function closeConfigDeclining(id) {
    return {
        type: CLOSE_CONFIG_DECLINING,
        id
    };
}

