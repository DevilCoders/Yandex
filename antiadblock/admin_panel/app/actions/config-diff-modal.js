export const OPEN_CONFIG_DIFF_MODAL = 'OPEN_CONFIG_DIFF_MODAL';
export const CLOSE_CONFIG_DIFF_MODAL = 'CLOSE_CONFIG_DIFF_MODAL';

export function openConfigDiffModal(serviceId, id, secondId) {
    return {
        type: OPEN_CONFIG_DIFF_MODAL,
        serviceId,
        id,
        secondId
    };
}

export function closeConfigDiffModal() {
    return {
        type: CLOSE_CONFIG_DIFF_MODAL
    };
}
