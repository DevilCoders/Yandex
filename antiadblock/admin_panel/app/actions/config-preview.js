export const OPEN_CONFIG_PREVIEW = 'OPEN_CONFIG_PREVIEW';
export const CLOSE_CONFIG_PREVIEW = 'CLOSE_CONFIG_PREVIEW';

export function openConfigPreview(serviceId, id, configData) {
    return {
        type: OPEN_CONFIG_PREVIEW,
        serviceId,
        id,
        configData
    };
}

export function closeConfigPreview(id) {
    return {
        type: CLOSE_CONFIG_PREVIEW,
        id
    };
}
