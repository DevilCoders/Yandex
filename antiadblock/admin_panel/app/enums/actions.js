export const ACTIONS = {
    // Временно скрыт мониторинг на сервисах, т.к. он пока не работает
    // GRAPHS: 'graphs',
    CONFIGS: 'configs',
    AUDIT: 'audit',
    INFO: 'info',
    UTILS: 'utils',
    // ANTIADB-2636
    // HEALTH: 'health',
    RULES: 'rules',
    SBS_RESULT_CHECKS: 'sbs-result-checks'
};

export const SECRET_ACTIONS = {
    SCREENSHOTS: 'screenshots-checks',
    PROFILES: 'sbs-profiles',
    LOGS: 'logs'
};
