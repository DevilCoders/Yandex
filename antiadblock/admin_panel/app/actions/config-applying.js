import configApi from 'app/api/config';
import {setGlobalErrors} from 'app/actions/errors';

export const OPEN_CONFIG_APPLYING = 'OPEN_CONFIG_APPLYING';
export const CLOSE_CONFIG_APPLYING = 'CLOSE_CONFIG_APPLYING';
export const START_ACTIVE_CONFIG_LOADING = 'START_ACTIVE_CONFIG_LOADING';
export const END_ACTIVE_CONFIG_LOADING = 'END_ACTIVE_CONFIG_LOADING';

export function openConfigApplying(serviceId, id, configData, target) {
    return {
        type: OPEN_CONFIG_APPLYING,
        serviceId,
        id,
        configData,
        target
    };
}

export function closeConfigApplying(id) {
    return {
        type: CLOSE_CONFIG_APPLYING,
        id
    };
}

export function fetchCurrentConfig(labelId, target) {
    return dispatch => {
        dispatch(startActiveConfigLoading(labelId, target));

        return configApi.fetchCurrentConfig(labelId, target).then(config => {
            return dispatch(endActiveConfigLoading(labelId, target, config));
        }, error => {
            return dispatch(setGlobalErrors([error.message]));
        });
    };
}

export function startActiveConfigLoading(labelId, target) {
    return {
        type: START_ACTIVE_CONFIG_LOADING,
        labelId,
        target
    };
}

export function endActiveConfigLoading(labelId, target, config) {
    return {
        type: END_ACTIVE_CONFIG_LOADING,
        labelId,
        target,
        config
    };
}
