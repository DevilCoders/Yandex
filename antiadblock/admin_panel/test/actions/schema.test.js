import * as schemaActions from 'app/actions/schema';
import * as schemaActionsFixtures from '../fixtures/actions/schema';

import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import {SCHEMA} from '../fixtures/backend/schema';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('schema actions', () => {
    const labelId = 'test.service';

    it('should create START_SCHEMA_LOADING', () => {
        expect(schemaActions.startSchemaLoading(labelId)).toEqual(schemaActionsFixtures.START_SCHEMA_LOADING);
    });

    it('should create END_SCHEMA_LOADING', () => {
        expect(schemaActions.endSchemaLoading(labelId, SCHEMA)).toEqual(schemaActionsFixtures.END_SCHEMA_LOADING);
    });

    it('should create START_SCHEMA_LOADING after loading', () => {
        const store = mockStore({});

        fetch.mockResponseOnce(JSON.stringify(SCHEMA));

        return store.dispatch(schemaActions.fetchSchema(labelId)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        });
    });

    it('should create SET_ERRORS after catched an error', () => {
        const store = mockStore({});

        fetch.mockRejectOnce(new Error('Test error'));

        return store.dispatch(schemaActions.fetchSchema(labelId)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        });
    });
});
