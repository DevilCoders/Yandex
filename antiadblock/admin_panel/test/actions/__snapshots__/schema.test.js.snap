// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`schema actions should create SET_ERRORS after catched an error 1`] = `
Array [
  Object {
    "labelId": "test.service",
    "type": "START_SCHEMA_LOADING",
  },
  Object {
    "errors": Array [
      "Произошла ошибка. Попробуйте ещё раз.",
    ],
    "id": "global",
    "type": "SET_ERRORS",
  },
]
`;

exports[`schema actions should create START_SCHEMA_LOADING after loading 1`] = `
Array [
  Object {
    "labelId": "test.service",
    "type": "START_SCHEMA_LOADING",
  },
  Object {
    "labelId": "test.service",
    "schema": Array [
      Object {
        "group_name": "TESTGROUP",
        "hint": "testgroup-hint",
        "items": Array [
          Object {
            "key": Object {
              "default": Array [],
              "name": "ACCEL_REDIRECT_URL_RE",
              "type_schema": Object {
                "children": Object {
                  "placeholder": "regexp-placeholder",
                  "type": "regexp",
                },
                "hint": "accel-redirect-url-re-hint",
                "title": "accel-redirect-url-re-title",
                "type": "array",
              },
            },
          },
          Object {
            "key": Object {
              "default": true,
              "name": "ADB_ENABLED",
              "type_schema": Object {
                "hint": "adb-enabled-hint",
                "placeholder": Object {
                  "placeholder_off": "adb-enabled-off-placeholder",
                  "placeholder_on": "adb-enabled-on-placeholder",
                },
                "title": "adb-enabled-title",
                "type": "bool",
              },
            },
          },
          Object {
            "key": Object {
              "default": false,
              "name": "ADFOX_DEBUG",
              "type_schema": Object {
                "hint": "adfox-debug-hint",
                "placeholder": Object {
                  "placeholder_off": "adfox-debug-off-placeholder",
                  "placeholder_on": "adfox-debug-on-placeholder",
                },
                "title": "adfox-debug-title",
                "type": "bool",
              },
            },
          },
          Object {
            "key": Object {
              "default": Array [],
              "name": "CLASSES_FOR_REPLACE",
              "type_schema": Object {
                "children": Object {
                  "placeholder": "classes-for-replace-placeholder",
                  "type": "string",
                },
                "hint": "classes-for-replace-hint",
                "title": "classes-for-replace-title",
                "type": "array",
              },
            },
          },
          Object {
            "key": Object {
              "default": Array [],
              "name": "CLIENT_REDIRECT_URL_RE",
              "type_schema": Object {
                "children": Object {
                  "placeholder": "regexp-placeholder",
                  "type": "regexp",
                },
                "hint": "client-redirect-url-re-hint",
                "title": "client-redirect-url-re-title",
                "type": "array",
              },
            },
          },
          Object {
            "key": Object {
              "default": "https://statchecker.yandex.ru/mapuid/",
              "name": "CM_IMAGE_URL",
              "type_schema": Object {
                "hint": "cm-image-url-hint",
                "placeholder": "link-placeholder",
                "title": "cm-image-url-title",
                "type": "string",
              },
            },
          },
          Object {
            "key": Object {
              "default": "//an.yandex.ru/mapuid/",
              "name": "CM_REDIRECT_URL",
              "type_schema": Object {
                "hint": "cm-redirect-url-hint",
                "placeholder": "link-placeholder",
                "title": "cm-redirect-url-title",
                "type": "string",
              },
            },
          },
          Object {
            "key": Object {
              "default": 3,
              "name": "CM_TYPE",
              "type_schema": Object {
                "hint": "cm-type-hint",
                "placeholder": "cm-type-placeholder",
                "title": "cm-type-title",
                "type": "select",
                "values": Object {
                  "cookiematching-disabled-placeholder": 0,
                  "cookiematching-image-or-refresh-placeholder": 3,
                  "cookiematching-image-placeholder": 1,
                  "cookiematching-refresh-placeholder": 2,
                },
              },
            },
          },
          Object {
            "key": Object {
              "default": null,
              "name": "CRYPTED_HOST",
              "type_schema": Object {
                "hint": "crypted-host-hint",
                "placeholder": "crypted-host-placeholder",
                "title": "crypted-host-title",
                "type": "string",
              },
            },
          },
          Object {
            "key": Object {
              "default": false,
              "name": "CRYPTED_URL_LENGTH_PREFIX",
              "type_schema": Object {
                "hint": "crypted-url-length-prefix-hint",
                "placeholder": Object {
                  "placeholder_off": "crypted-url-length-prefix-off-placeholder",
                  "placeholder_on": "crypted-url-length-prefix-on-placeholder",
                },
                "title": "crypted-url-length-prefix-title",
                "type": "bool",
              },
            },
          },
          Object {
            "key": Object {
              "default": Object {},
              "name": "CRYPT_BODY_RE",
              "type_schema": Object {
                "hint": Object {
                  "crypt_hint": "crypt-body-re-crypt-hint",
                  "substitute_hint": "crypt-body-re-substitute-hint",
                },
                "placeholder": Object {
                  "crypt_placeholder": "regexp-placeholder",
                  "replaceable_placeholder": "regexp-placeholder",
                  "substitute_placeholder": "crypt-body-re-substitute-placeholder",
                },
                "title": Object {
                  "crypt_title": "crypt-body-re-crypt-title",
                  "substitute_title": "crypt-body-re-substitute-title",
                },
                "type": "crypt_body",
              },
            },
          },
          Object {
            "key": Object {
              "default": false,
              "name": "CRYPT_ENABLE_TRAILING_SLASH",
              "type_schema": Object {
                "hint": "crypt-enable-trailing-slash-hint",
                "placeholder": Object {
                  "placeholder_off": "crypt-enable-trailing-slash-off-placeholder",
                  "placeholder_on": "crypt-enable-trailing-slash-on-placeholder",
                },
                "title": "crypt-enable-trailing-slash-title",
                "type": "bool",
              },
            },
          },
          Object {
            "key": Object {
              "default": Array [],
              "name": "CRYPT_RELATIVE_URL_RE",
              "type_schema": Object {
                "children": Object {
                  "placeholder": "regexp-placeholder",
                  "type": "string",
                },
                "hint": "crypt-relative-url-re-hint",
                "title": "crypt-relative-url-re-title",
                "type": "array",
              },
            },
          },
          Object {
            "key": Object {
              "default": null,
              "name": "CRYPT_SECRET_KEY",
              "required": true,
              "type_schema": Object {
                "hint": "crypt-secret-key-hint",
                "placeholder": "crypt-secret-key-placeholder",
                "title": "crypt-secret-key-title",
                "type": "string",
              },
            },
          },
          Object {
            "key": Object {
              "default": "/",
              "name": "CRYPT_URL_PREFFIX",
              "type_schema": Object {
                "hint": "crypt-url-preffix-hint",
                "placeholder": "crypt-url-preffix-placeholder",
                "title": "crypt-url-preffix-title",
                "type": "string",
              },
            },
          },
          Object {
            "key": Object {
              "default": Array [],
              "name": "CRYPT_URL_RE",
              "required": true,
              "type_schema": Object {
                "children": Object {
                  "placeholder": "regexp-placeholder",
                  "type": "regexp",
                },
                "hint": "crypt-url-re-hint",
                "title": "crypt-url-re-title",
                "type": "array",
              },
            },
          },
          Object {
            "key": Object {
              "default": Array [],
              "name": "DETECT_COOKIE_DOMAINS",
              "type_schema": Object {
                "children": Object {
                  "placeholder": "detect-cookie-domains-placeholder",
                  "type": "string",
                },
                "hint": "detect-cookie-domains-hint",
                "title": "detect-cookie-domains-title",
                "type": "array",
              },
            },
          },
          Object {
            "key": Object {
              "default": 0,
              "name": "DETECT_COOKIE_TYPE",
              "type_schema": Object {
                "hint": "detect-cookie-type-hint",
                "placeholder": "detect-cookie-type-placeholder",
                "title": "detect-cookie-type-title",
                "type": "select",
                "values": Object {
                  "detect-cookie-type-children-placeholder": 1,
                  "detect-cookie-type-current-placeholder": 0,
                  "detect-cookie-type-list-placeholder": 2,
                },
              },
            },
          },
          Object {
            "key": Object {
              "default": Array [],
              "name": "DETECT_HTML",
              "type_schema": Object {
                "children": Object {
                  "placeholder": "detect-html-placeholder",
                  "type": "detect_html",
                },
                "hint": "detect-html-hint",
                "title": "detect-html-title",
                "type": "array",
              },
            },
          },
          Object {
            "key": Object {
              "default": Array [],
              "name": "DETECT_LINKS",
              "type_schema": Object {
                "children": Object {
                  "placeholder": "link-placeholder",
                  "type": "link",
                },
                "hint": "detect-links-hint",
                "title": "detect-links-title",
                "type": "array",
              },
            },
          },
          Object {
            "key": Object {
              "default": Array [
                0,
              ],
              "name": "ENCRYPTION_STEPS",
              "required": true,
              "type_schema": Object {
                "hint": "encryption-steps-hint",
                "placeholder": "encryption-steps-placeholder",
                "title": "encryption-steps-title",
                "type": "multi_select",
                "values": Object {
                  "encryption-steps-advertising-crypt-placeholder": 1,
                  "encryption-steps-partner-content-class-crypt-placeholder": 3,
                  "encryption-steps-partner-content-url-crypt-placeholder": 2,
                  "encryption-steps-pcode-replace-placeholder": 0,
                  "encryption-steps-tags-crypt-placeholder": 4,
                },
              },
            },
          },
          Object {
            "key": Object {
              "default": Array [
                "bltsr",
              ],
              "name": "EXCLUDE_COOKIE_FORWARD",
              "type_schema": Object {
                "children": Object {
                  "placeholder": "exclude-cookie-forward-placeholder",
                  "type": "string",
                },
                "hint": "exclude-cookie-forward-hint",
                "title": "exclude-cookie-forward-title",
                "type": "tags",
              },
            },
          },
          Object {
            "key": Object {
              "default": Array [
                "addruid",
              ],
              "name": "EXTUID_COOKIE_NAMES",
              "type_schema": Object {
                "children": Object {
                  "placeholder": "extuid-cookie-names-placeholder",
                  "type": "string",
                },
                "hint": "extuid-cookie-names-hint",
                "title": "extuid-cookie-names-title",
                "type": "tags",
              },
            },
          },
          Object {
            "key": Object {
              "default": null,
              "name": "EXTUID_TAG",
              "required": true,
              "type_schema": Object {
                "hint": "extuid-tag-hint",
                "placeholder": "extuid-tag-placeholder",
                "title": "extuid-tag-title",
                "type": "string",
              },
            },
          },
          Object {
            "key": Object {
              "default": Array [],
              "name": "FOLLOW_REDIRECT_URL_RE",
              "type_schema": Object {
                "children": Object {
                  "placeholder": "regexp-placeholder",
                  "type": "regexp",
                },
                "hint": "follow-redirect-url-re-hint",
                "title": "follow-redirect-url-re-title",
                "type": "array",
              },
            },
          },
          Object {
            "key": Object {
              "default": Object {},
              "name": "HOSTNAME_MAPPING",
              "type_schema": Object {
                "hint": "hostname-mapping-hint",
                "key": "string",
                "placeholder": Object {
                  "key_placeholder": "hostname-mapping-key-placeholder",
                  "value_placeholder": "hostname-mapping-value-placeholder",
                },
                "title": "hostname-mapping-title",
                "type": "object",
                "value": "string",
              },
            },
          },
          Object {
            "key": Object {
              "default": false,
              "name": "INTERNAL",
              "type_schema": Object {
                "hint": "internal-hint",
                "placeholder": Object {
                  "placeholder_off": "internal-off-placeholder",
                  "placeholder_on": "internal-on-placeholder",
                },
                "title": "internal-title",
                "type": "bool",
              },
            },
          },
          Object {
            "key": Object {
              "default": Array [],
              "name": "PARTNER_TOKENS",
              "required": true,
              "type_schema": Object {
                "children": Object {
                  "placeholder": "partner-tokens-placeholder",
                  "type": "string",
                },
                "hint": "partner-tokens-hint",
                "title": "partner-tokens-title",
                "type": "array",
              },
            },
          },
          Object {
            "key": Object {
              "default": Array [],
              "name": "PROXY_URL_RE",
              "required": true,
              "type_schema": Object {
                "children": Object {
                  "placeholder": "regexp-placeholder",
                  "type": "regexp",
                },
                "hint": "proxy-url-re-hint",
                "title": "proxy-url-re-title",
                "type": "array",
              },
            },
          },
          Object {
            "key": Object {
              "default": null,
              "name": "PUBLISHER_SECRET_KEY",
              "required": true,
              "type_schema": Object {
                "hint": "publisher-secret-key-hint",
                "placeholder": "publisher-secret-key-placeholder",
                "title": "publisher-secret-key-title",
                "type": "string",
              },
            },
          },
          Object {
            "key": Object {
              "default": null,
              "name": "TEST_DATA",
              "type_schema": Object {
                "hint": Object {
                  "invalid_hint": "test-data-hint",
                  "valid_hint": "test-data-hint",
                },
                "placeholder": Object {
                  "invalid_placeholder": "test-data-placeholder",
                  "valid_placeholder": "test-data-placeholder",
                },
                "title": Object {
                  "invalid_title": "test-data-invalid-title",
                  "valid_title": "test-data-valid-title",
                },
                "type": "test_data",
              },
            },
          },
        ],
        "title": "testgroup-title",
      },
    ],
    "type": "END_SCHEMA_LOADING",
  },
]
`;
